### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by timomitchel**
Added 0 or operator to add item and fixed cart show logic

### CODE REVIEW

**Comment by timomitchel** [comment]
@AliSchlereth Hey Ali we're wondering about where and when we should be clearing our session[:cart]. We did this within byebug, but we have not implemented it in our controller yet. We added some functionality to our cart show view and wrote a test that is working for one item, but we ran into some issues trying to add multiple items to our cart. We ended up leaving that alone and following the user stories, but we were hoping to chat about a strategy to implement adding multiple items. Thanks so much for the input!

**Comment by AliSchlereth** [comment]
@timomitchel  I'd want the cart to clear when a user logs out, the order is completed, or sessions/cookies are cleared.

File: app/models/cart.rb [hunk 1]
```diff
@@ -2,23 +2,31 @@ class Cart
...
+    @contents = contents || Hash.new(0)
   end
 
   def total_count
     contents.values.sum
   end
 
   def add_item(id)
-    contents[id.to_s] += 1
+    contents[id.to_s] = (contents[id.to_s] || 0) + 1
```
Comment (AliSchlereth) [inline app/models/cart.rb:14]: I'm curious why you switched this from `contents[id.to_s] += 1`?

File: config/routes.rb [hunk 2]
```diff
@@ -5,12 +5,14 @@
...
   get "/dashboard", to: "users#show"
+  get '/cart', to: "carts#show"
+  delete '/carts', to: "carts#destroy"
 
   resources :users, only:[:new, :create, :show]
 
   resources :items, only: [:index]
 
-  resources :carts, only: [:create]
+  resources :carts, only: [:create, :show]
```
Comment (AliSchlereth) [inline config/routes.rb:15]: This feels like you have some duplication between lines 9 and 15.

**Comment by AliSchlereth** [comment]
@timomitchel Added some further comments.

**Pull request merged**